### PR TITLE
MWPW-127736 Fix Safari sign-in dropdown

### PR DESCRIPTION
**Pull request by meganthecoder**
* Fix issue where clicking "Sign in" on safari on mobile device didn't show dropdown
* Fix issue where "Sign in" button is inaccessible on Safari 14 mobile
* Fix issue where "Sign in" dropdown and toggle/hamburger menu can be open at the same time

Resolves: [MWPW-127736](https://jira.corp.adobe.com/browse/MWPW-127736)

**Test URLs:**
- Before: https://main--milo--adobecom.hlx.page/drafts/methomas/checkmark?martech=off
- After: https://methomas-sign-in-mobile--milo--adobecom.hlx.page/drafts/methomas/checkmark?martech=off

BACOM:
- Before: https://main--bacom--adobecom.hlx.page/customer-success-stories?martech=off
- After: https://main--bacom--adobecom.hlx.page/customer-success-stories?milolibs=methomas-sign-in-mobile&martech=off

Note: This only occurs on iOS, so you'll need to use browserstack to verify.

### CODE REVIEW

**Review by JasonHowellSlavin — APPROVED** [review]

**Comment by JasonHowellSlavin** [comment]
I think this fixes the ticket as written, but in the second video entitled "[Expected] Multicloud..." when the hamburger menu is open and a user clicks "Sign In" it closes the hamburger menu and vice versa. Might be good to double back with Malik.

**Review by sukamat — APPROVED** [review]

**Review by chrischrischris — APPROVED** [review]

**Comment by meganthecoder** [comment]
As per Jason's comment, I discussed with Malik and the sign-in dropdown needs to close when you click the hamburger menu. Working on that update

File: libs/blocks/gnav/gnav.js [hunk 1]
```diff
@@ -117,18 +117,31 @@ class Gnav {
...
+      }
+    };
     const onMediaChange = (e) => {
       if (e.matches) {
         this.el.classList.remove(IS_OPEN);
+        document.removeEventListener('click', closeToggleOnDocClick);
       }
     };
-    toggle.addEventListener('click', async () => {
+    toggle.addEventListener('click', async (e) => {
```
Comment (sukamat) [inline libs/blocks/gnav/gnav.js:133]: `e` doesn't seem to be used here.

Comment (meganthecoder) [reply libs/blocks/gnav/gnav.js]: Good catch!

**Review by sukamat — APPROVED** [review]

**Review by JasonHowellSlavin — APPROVED** [review]
LGTM!

**Review by Brandon32 — APPROVED** [review]

**Review by honstar — APPROVED** [review]
LGTM

**Review by chrischrischris — APPROVED** [review]

**Pull request merged**